### PR TITLE
docker_container: add device_requests option

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes ansible/ansible#65748.

Unfortunately it is hard to test this in CI, since we don't have special hardware and docker runtimes installed. The tests are therefore very minimalistic and don't really test something.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
docker_container

### CODE REVIEW

**Comment by felixfontein** [comment]
CC @ei-grad @ssidorenko @mbigras @pete0emerson @Samh57 @xywsxp @tizz98 @bingzhangdai @pommedeterresautee since you all liked ansible/ansible#65748 or https://github.com/ansible/ansible/issues/65748#issuecomment-671771057. Please test this, since I don't really have the ability to do so, and I guess you have :)

File: plugins/modules/cloud/docker/docker_container.py [hunk 1]
```diff
@@ -212,6 +212,40 @@
...
+      - Allows to request additional resources, such as GPUs.
+    type: list
+    elements: dict
+    suboptions:
+      capabilities:
+        description:
+          - List of lists of strings to request capabilities.
+          - The top-level list entries are combined by OR, and for every list entry,
+            the entries in the list it contains are combined by AND.
+          - The driver will try to satisfy one of the sub-lists.
```
Comment (Andersson007) [inline plugins/modules/cloud/docker/docker_container.py:226]:
```suggestion
          - The driver will try to satisfy one of the sub-lists.
```
would be good to rephrase this to use a present tense as recommended in best practices.

**Review by Andersson007 — COMMENTED** [review]
Would be also good to have:
1) examples for new options
2) `check_mode: yes` in the tests

File: plugins/modules/cloud/docker/docker_container.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/cloud/docker/docker_container.py]: Done in 36b2645db46ecc19adde875b1f2d85dd2b1995a9.

**Comment by felixfontein** [comment]
I've added an example. I won't add check_mode tests for this option, since we don't do it for any other option either, as it would put a too large strain on CI resources. The docker_container tests are one of the most extensive tests already (over 650 tests) and run for 5-7 minutes (not counting setup). Also, the way the module is implemented makes additional check_mode tests for every option superfluous :)

**Review by Andersson007 — APPROVED** [review]
the fragment, example, and tests LGTM:)
